File: src/worker.py
"""Worker entrypoint: the same `app` inside workerd, backed by D1.

Why this file is separate from main.py: the Workers runtime executes the
entrypoint's top-level scope at deploy time and takes a memory snapshot,
so this module must stay side-effect free. `asgi.entrypoint` wires the
JS `Request` into the ASGI app and puts the bindings object on
`request.scope["env"]`, which the request dependency in db.py reads.
"""

from workers import asgi

from main import app

Default = asgi.entrypoint(app)

File: src/worker.ts
/**
 * Worker entrypoint: the Hono app itself, served by workerd.
 *
 * WHY the app is a valid module Worker as-is: a Hono instance exposes a
 * `fetch(request, env, ctx)` method, which is exactly workerd's ES module
 * handler contract. The default export is the whole entrypoint; no
 * adapter, no bridge. Unlike the Pyodide era's `asgi.entrypoint`, the
 * top-level scope has no snapshot-safety constraint beyond "no I/O at
 * import time".
 */

import { app } from "./main";

export default app;
